### PR TITLE
CNI mode: validate initial pod list

### PR DESCRIPTION
**Pull request by universam1**
There is a corner case where the initial listing of the pods happens during an ingress pod start phase. A pod without an assigned IP address might be included then, causing an ALB TG update error.

We are validating the pod statuses in the informer event handler but not yet at the initial listing. Adding the same validations there as well.

### CODE REVIEW

**Comment by universam1** [comment]
@AlexanderYastrebov @szuecs

File: kubernetes/pods.go [hunk 1]
```diff
@@ -43,7 +43,9 @@ func (a *Adapter) PodInformer(ctx context.Context, endpointChan chan<- []string)
 		time.Sleep(resyncInterval)
 	}
 	for _, pod := range podList {
-		podEndpoints.Store(pod.Name, pod.Status.PodIP)
+		if isPodRunning(pod) {
```
Comment (AlexanderYastrebov) [inline kubernetes/pods.go:46]: Can we get terminating pod here as well?

Comment (AlexanderYastrebov) [reply kubernetes/pods.go:46]: Ah, nevermind, we only store running pods here

**Comment by universam1** [comment]
> Add a test for it maybe?

NP, covered by TestAdapter_PodInformer and explicitly now

File: kubernetes/pods.go [hunk 2]
```diff
@@ -43,7 +43,9 @@ func (a *Adapter) PodInformer(ctx context.Context, endpointChan chan<- []string)
...
+			podEndpoints.Store(pod.Name, pod.Status.PodIP)
+		}
```
Comment (AlexanderYastrebov) [inline kubernetes/pods.go:48]: > Add a test for it maybe?
>> NP, covered by TestAdapter_PodInformer and explicitly now

Well, I meant the listing part - the filtering was introduced but no test was changed which means it was not covered.

Comment (universam1) [reply kubernetes/pods.go]: added - the test should contain all conditions now

**Comment by universam1** [comment]
@szuecs would be good to get this one out the door asap

**Comment by AlexanderYastrebov** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**